src/libraries/System.Security.Cryptography.Csp/src/System/Security/Cryptography/DSACryptoServiceProvider.Unix.cs

```diff
@@ -147,14 +161,14 @@ public override int KeySize
             set
             {
                 // Perform the check here because LegalKeySizes are more restrictive here than _impl
-                if (!value.IsLegalSize(s_legalKeySizes))
+                if (!value.IsLegalSize(this.LegalKeySizesValue!))
```

Comment:
The `this.` shouldn't be required.  Also, if you save the result of the OS-based ternary to `s_legalKeySizes`, this line can be reverted.